hls4ml/writer/vivado_accelerator_writer.py

```diff
@@ -348,14 +416,129 @@ def write_board_script(self, model):
-        copyfile(os.path.join(filedir, self.vivado_accelerator_config.get_driver_path()),
-                 ('{}/' + self.vivado_accelerator_config.get_driver_file()).format(model.config.get_output_dir()))
-        
+        srcfiles = os.path.join(filedir, self.vivado_accelerator_config.get_driver_path())
+        dstfiles = ('{}/' + self.vivado_accelerator_config.get_driver_files()).format(model.config.get_output_dir())
+        if os.path.isdir(srcfiles):
+            copytree(srcfiles, dstfiles, dirs_exist_ok=True)
+        else:
+            copyfile(srcfiles, dstfiles)
+
     def write_new_tar(self, model):
         os.remove(model.config.get_output_dir() + '.tar.gz')
         super(VivadoAcceleratorWriter, self).write_tar(model)
 
-        
+    def write_standalone_app(self, model):
+
+        indent = '    '
+
+        weights = model.get_weight_variables()
+
+        io_type = model.config.get_config_value('IOType')
+        interface = model.config.get_config_value('AcceleratorConfig')['Interface'] if model.config.get_config_value('AcceleratorConfig') else None
+        config_weights = (io_type == 'io_stream') and (interface == 'axi_master')
+
+        #######################
+        ## main.c
+        #######################
+
+        filedir = os.path.dirname(os.path.abspath(__file__))
+        f = open(os.path.join(filedir, '../templates/vivado_accelerator/standalone_main.c'), 'r')
+        fout = open('{}/sdk/common/main.c'.format(model.config.get_output_dir()), 'w')
+
+        for line in f.readlines():
+
+            if config_weights and '/*hls-fpga-machine-learning insert configure weights*/' in line:
+                newline = line
+                for w in weights:
+                    newline += indent + 'XMyproject_axi_Set_{name}(&accelerator, {name}); /* TODO: design-dependent name */\n'.format(name=w.name)
+            elif config_weights and '/*hls-fpga-machine-learning insert load weights on*/' in line:
+                newline = line
+                newline += indent + 'XMyproject_axi_Set_load_weights(&accelerator, 1); /* TODO: design-dependent name */\n'
+            elif config_weights and '/*hls-fpga-machine-learning insert load weights off*/' in line:
+                newline = line
+                newline += indent + indent + 'XMyproject_axi_Set_load_weights(&accelerator, 0); /* TODO: design-dependent name */\n'
+            elif config_weights and '/*hls-fpga-machine-learning insert start and wait*/' in line:
+                newline = line
+                newline += indent + 'XMyproject_axi_Start(&accelerator); /* TODO: design-dependent name */\n'
+                newline += indent + 'while (!XMyproject_axi_IsDone(&accelerator)); /* TODO: design-dependent name */\n'
+            else:
+                newline = line
+            fout.write(newline)
+
+        f.close()
+        fout.close()
+
+    def write_header_file(model, X, y, y_keras, y_hls, n_samples, filename='data.h'):
```

Comment:
Related to the comment above - maybe this should be an example?

Reply:
I am good with moving those functions to the examples, and later on, we can "promote" them to the hls4ml codebase if necessary.